Automattic/jetpack
# Add/bad installation notices to my jetpack

## Proposed changes:

* Show red bubble notification for bad installs by using the filter on each plugin during the installation check
* Combine `useConnectionWatcher` with the new bad install notices
* Display bad install notices as MEDIUM priority on My Jetpack

Note: This PR contains the changes made in https://github.com/Automattic/jetpack/pull/36438 for notice priority. I will rebase this after that one is merged.

### Other information:


## Jetpack product discussion

pbNhbs-aby-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. If you know how to simulate a bad plugin installation (or actually install a plugin incorrectly) you can test it that way. But the following instructions will be for testing this on your local environment
2. Checkout this branch and run `jetpack install packages/my-jetpack` and `jetpack build packages/my-jetpack`. You shouldn't need to rebuild the plugins since the changes are on the server-side, but you might try that if this doesn't work for you
3. Make sure your site is connected
4. Activate the Jetpack Boost standalone plugin on your site (so you can still see My Jetpack if Jetpack is installed incorrectly)
5. Go to `projects/plugins/jetpack/jetpack.php` and edit line 143 to be
```php
if ( ! is_readable( $jetpack_autoloader ) && is_readable( $jetpack_module_headings_file ) ) {
```
6. Go to `/wp-admin/admin.php?page=my-jetpack` and make sure the bubble notification is active and that you see the error notice in My Jetpack. Make sure the CTA works correctly.
![image](https://github.com/Automattic/jetpack/assets/65001528/0592a73d-f112-4189-9325-8bacc71d6b9e)
7. Activate Jetpack Search on your site and go to `projects/plugins/search/jetpack-search.php` and update line 47 to be 
```php
if ( is_readable( $autoload_packages_path ) ) {
```
8. Go to `/wp-admin/admin.php?page=my-jetpack` and make sure the bubble notification now shows 2 errors, but that only one notice is showing in My Jetpack
![image](https://github.com/Automattic/jetpack/assets/65001528/927fdfca-0c61-4c1e-8ab6-ab3f1b38f75a)
9. Disconnect your site and make sure the site connection notice shows instead of the bad installation notices, and that the red bubble notification shows 3 notifications now
![image](https://github.com/Automattic/jetpack/assets/65001528/62f8bf62-97fc-44bf-8dc6-38a3c57bbcae)


Bonus points: Try any number of combination of bad install, site connection, or failed API call notices and make sure everything works as you would expect
